core/domain/exp_domain.py

```diff
@@ -534,25 +534,21 @@ def validate(self, obj_type, exp_param_specs_dict):
                 'There must be at least one rule for each answer group.'
                 % self.rule_specs)
 
-        all_rule_classes = rule_domain.get_rules_for_obj_type(obj_type)
         seen_fuzzy_rule = False
         for rule_spec in self.rule_specs:
-            rule_class = None
-            try:
-                rule_class = next(
-                    r for r in all_rule_classes
-                    if r.__name__ == rule_spec.rule_type)
-            except StopIteration:
+            if rule_spec.rule_type not in interaction_rules:
                 raise utils.ValidationError(
                     'Unrecognized rule type: %s' % rule_spec.rule_type)
-            if rule_class.__name__ == rule_domain.FUZZY_RULE_TYPE:
+
+            if rule_spec.rule_type == rule_domain.FUZZY_RULE_TYPE:
                 if seen_fuzzy_rule:
                     raise utils.ValidationError(
                         'AnswerGroups can only have one fuzzy rule.')
                 seen_fuzzy_rule = True
 
             rule_spec.validate(
-                rule_domain.get_param_list(rule_class.description),
+                rule_domain.get_param_list(
+                    interaction_rules[rule_spec.rule_type]['description']),
```

Comment:
Probably should use ``get_rule_description()`` here for consistency (and, hence, pass in interaction into this validate() instead).